crates/turborepo-auth/src/auth/logout.rs

```diff
@@ -1,5 +1,34 @@
-use turborepo_ui::{cprintln, GREY, UI};
+use std::fs;
 
-pub fn logout(ui: &UI) {
-    cprintln!(ui, GREY, ">>> Logged out");
+use tracing::error;
+use turborepo_api_client::Client;
+use turborepo_ui::{cprintln, GREY};
+
+use crate::{Error, LogoutOptions};
+
+pub fn logout<T: Client>(options: &LogoutOptions<T>) -> Result<(), Error> {
+    if let Err(err) = remove_token(options) {
+        error!("could not logout. Something went wrong: {}", err);
+        return Err(err);
+    }
+
+    cprintln!(options.ui, GREY, ">>> Logged out");
+    Ok(())
+}
+
+fn remove_token<T: Client>(options: &LogoutOptions<T>) -> Result<(), Error> {
+    let global_config_path = options.path;
+    let content =
+        fs::read_to_string(global_config_path).map_err(|e| Error::FailedToReadAuthFilePath {
+            path: global_config_path.clone(),
+            error: e,
+        })?;
+
+    let mut data: serde_json::Value = serde_json::from_str(&content)?;
+    data["token"] = serde_json::Value::Null;
+
+    let new_content = serde_json::to_string_pretty(&data)?;
+    fs::write(global_config_path, new_content)?;
```

Comment:
If we want to preserve previous behavior we should keep using `unset_path`. `serde` will reformat/reorder the contents of the JSON.
```suggestion
    if let Some(after) = unset_path(&before, &["token"], true)? {
        global_config_path.create_with_contents(after).map_err(|e| {
            Error::Config(config::Error::FailedToSetConfig {
                config_path: global_config_path.clone(),
                error: e,
            })
        })
    }
```

Reply:
the problem with `unset_path` is that it lives in the crate that will call this one. I can move this same logic or similar logic into this crate or create a new one for file interactions, but otherwise this causes a cyclic import which makes Rust and myself sad. I could be importing incorrectly and maybe I can _just_ grab this module, but it still feels a little odd.

As for the contents, AFAIK we don't store anything _but_ the token in the `config.json`, so ordering / formatting won't be an issue at the moment. If we add more to the JSON then this would need to change, so I'm open to doing some of the previous suggestions.